Life.java

```diff
@@ -1,55 +1,101 @@
 public class Life implements ILife {
+  int width;
+  int height;
+  int[][] board;
   
   public static void main(String[] args) {
-    Life l = new Life(new String[] {  "     ",
-                                      "     ",
-                                      " *** ",
-                                      "     ",
-                                      "     " });
-    l = (Life) l.nextGeneration();
   }
 
-
-  public Life() {
-    nukeAll();
+  public Life(int width, int height) {
+    this.width = width;
+    this.height = height;
+    this.board = new int[width][height];
   }
 
-  public Life(String[] setup) {
-    this();
-    for (int y = 0; y < setup.length; y++)
-      for (int x = 0; x < setup[y].length(); x++)
-        if (setup[y].charAt(x) != ' ')
-          setAlive(x, y);
+  // Just for testing the Graphical output of the Game of Life
+  public void printBoard(){
+    for(int y = 0; y < height; y++){
+      String line = "|";
+      for(int x = 0; x < width; x++){
+        if(this.board[x][y] == 0){
+          line += ".";
+        }else{
+          line += "*";
+        }
+      }
+      line += "|";
+      System.out.println(line);
+    }
+    System.out.println("\n");
   }
 
 
   @Override
   public void nukeAll() {
-    // TODO Auto-generated method stub
+    for(int y = 0; y < height; y++){
+      for(int x = 0; x < width; x++){
+        this.board[x][y] = 0;
+      }
+    }
 
   }
 
   @Override
   public void setAlive(int x, int y) {
-    // TODO Auto-generated method stub
-
+    this.board[x][y] = 1;
   }
 
   @Override
   public void setDead(int x, int y) {
-    // TODO Auto-generated method stub
-
+    this.board[x][y] = 0;
   }
 
   @Override
-  public boolean isAlive(int x, int y) {
-    // TODO Auto-generated method stub
-    return false;
+  public int isAlive(int x, int y) {
+    if(x < 0 || x >= width){
+      return 0;
+    }
+    if(y < 0 || y >= height){
+      return 0;
+    }
+    return this.board[x][y];
   }
 
   @Override
   public ILife nextGeneration() {
-    // TODO Auto-generated method stub
+    int[][] tempBoard = new int[width][height];
+    for(int y = 0; y < height; y++){
+      for (int x = 0; x < width; x++){
+        int aliveNeighbours = countAliveNeughbours(x, y);
+        if(this.board[x][y] == 1){
+          if(aliveNeighbours < 2){
+            tempBoard[x][y] = 0;
+          }else if(aliveNeighbours == 2 || aliveNeighbours == 3){
+            tempBoard[x][y] = 1;
+          }else if (aliveNeighbours > 3){
+            tempBoard[x][y] = 0;
+          }
+        }else{
+          if(aliveNeighbours == 3){
+            tempBoard[x][y] = 1;
+          }
+        }
+      }
+    }
+    this.board = tempBoard;
     return null;
   }
+
+  public int countAliveNeughbours(int x, int y){
+    int count = 0;
+    count += isAlive(x-1,y-1);
```

Comment:
hier sollten sie ihre Indizes prüfen, ob sie auch gültig sind!
Das funktioniert ja nur deshalb, weil sie in isAlive die Fehler ignorieren s.o.